src/JSInterop/Microsoft.JSInterop/src/Infrastructure/JSCallInfo.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.JSInterop.Infrastructure
+{
+    /// <summary>
+    /// Information about a JSInterop call from JavaScript to .NET.
+    /// </summary>
+    public readonly struct JSCallInfo
+    {
+        public JSCallInfo(string methodIdentifier, string assemblyName, long? dotNetObjectId, long? callId)
```

Comment:
I also tweaked this to use the correct representation for data  types. We were reading everything as a string today and then parsing them in to longs. Just feels wrong. It's fairly trivial to add support for `long?` to BlazorPack, so I changed these data types

Reply:
While we're here, the order in which these parameters arrive is kinda odd:

https://github.com/aspnet/Extensions/blob/master/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L62

`callId, assemblyName, methodIdentifier, dotNetObjectId, argsJson`

`callId` doesn't apply to sync code and may be null, `assemblyName` and `objectId` are discriminated (only one of the two can be specified). `methodIdentifier` is the only value that must always be present. I changed the order in the C# since it feels correct, but left it alone in TypeScript because it seems like unnecessary work.